src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -1217,6 +1233,7 @@ protected override void Go()
                             if (len > i && _operator == RegexCode.Notoneloop)
                             {
                                 TrackPush(len - i - 1, runtextpos - Bump());
+                                _maxBacktrackPosition = tempMaxBacktrackPosition;
```

Comment:
Is this correct even for right-to-left?

Reply:
It is yea. `runtextpos` will be at the right end initially for a right-to-left run and `.*` will move `runtextpos` to the left end. So `_maxBacktrackPosition = tempMaxBacktrackPosition = rightEnd` seems right to me.